### DISCUSSION TITLE
Cannot seem to add TPU filament to a Creality E3 S1 Pro profile

### DISCUSSION THREAD

**Discussion by The-Bishop82** [OrcaSlicer/OrcaSlicer#3947] [Q&A]
I've just been driving myself half-batty trying to add a TPU filament to my Ender 3 S1 Pro printer profile. 

I can see the generic ones in the profile for my K1.  I added an ender 3 V3 SE and I can see the TPU filament there. But it doesn't show under the Ender 3 S1 Pro. I can't even create a custom filament for the S1 Pro, anything I make shows up under the other printers.

Not sure if I'm doing something wrong, or there's some sort of odd bug.

**Comment by The-Bishop82** [comment]
I did a little more tinkering around. Added in an Ender 3 V3SE profile. Am able to see and use all the installed filaments. 
However, The Ender 3 S1 Pro (as well as the E 3 S1 non-pro) cannot see or use anything other than ASA/ABS, PLA, and PETG.  I can't even add a filament via 'custom'.

What I was able to do is save a custom version of TPU, then edit the json file to change the 'compatible_printers' value to match the printer name "Creality Ender-3 S1 Pro 0.4 nozzle" as well as the file names for the json and info file. Suddenly it showed as a usable filament for the S1 Pro printer.

Is it some kind of odd naming bug?

Edit: Found the filament profiles and added the S1 pro's printer name to the 'compatible printers' list for Creality TPU, now it shows up as a usable filament.

**Reply by Agent-047185** [reply]
@The-Bishop82 how did you managed to do that please? I found the same trick but everytime I restart Orca, the profile disappears from the json file. tried with wordpad and vscode...thank you

**Comment by mecgyver74** [comment]
Hi All, same printer, same problem. Unfortunately your workaround doesn't work for me.

Any way to force the printer to use filaments that are not included in the standard options? It seems that this operation is feasible for other printers, but impossible for the Creality Ender-3 S1 Pro!
Thank you
Andrea

**Comment by Vaerchi** [comment]
Same thing occurs with S1 Plus.  To add it, since I have sonicpad, i had to edit the Creality Generic TPU file in %appdata%\OrcaSlicer\system\Creality\filament.

**Reply by sinker1345** [reply]
Can confirm this worked fore after adding the tpu to the printer profile

**Reply by geshkin** [reply]
Hi, can you provide your TPU filament profile for S1 Plus?
Thanks.

**Comment by retirement2021** [comment]
looking for the same answer to this problem

**Comment by apalazzi** [comment]
Same issue here, I cannot add TPU to my elegoo Neptune 3 Pro.

**Comment by dockrbyter** [comment]
Same here....
https://github.com/SoftFever/OrcaSlicer/issues/3242

**Comment by Ergonomicmike** [comment]
As a workaround: I haven't found the need for a baked-in TPU filament profile. I simply modify both Filament and Process Presets with the values I need to print TPU. (This includes overriding some earlier settings. E.g., 8 mm/sec speed and I override retraction to zero.) I add the name "TPU" in their Preset names. Done.

(In fact, I don't use any of the baked-in Filament profiles in my normal work flow. I start with a standard profile as a baseline/Sanity Check, but the settings aren't what I want and I always have to modify the standard profiles for my brand, color and type of filament.)

Having said that, I have modified a json to change my travel acceleration. But it wasn't easy or obvious. There's a lot going on in those .json's with inheritances and everything has to match/be consistent for changes to take.

**Comment by fibr** [comment]
Experiencing the same problem
a) can't select TPU filament for S1 Pro (even after trying to edit the filament file as suggested above and restarting Orca)
b) can't add custom filaments. something about "you need to select a printer" but there's nowhere to select a printer
I did a workaround by saving a custom profile but the TPU results are pretty bad - so I guess a bunch of tuning is in order now. Would be really nice to just be able to use the TPU material

Suggestion: add a checkbox: "allow non-compatible filaments" to help with a better workaround

**Comment by sterkey25** [comment]
Got the workaround to work on windows but i need this to work on linux. Still broken...

**Comment by bsimmo** [comment]
So searching first would have seemed me quite some time.
Cannot select a TPU, cannot make a TPU from the add a new one.

How to we put a fix in for this printer?

**Comment by bsimmo** [comment]
#7085 Go there, add a thumbs up ?

**Comment by nambuco** [comment]
Same issue here, going back to prusa slicer until this issue is solved in Orca Slicer

**Comment by vechiato** [comment]
It worked for me on Orca 2.2.0 on a Mac. I had to add the printer at:  

`/Users/<USER>/Library/Application Support/OrcaSlicer/system/Creality/filament/Creality Generic TPU.json`  

under **compatible printer**:  

`"Creality Ender-3 S1 Pro 0.4 nozzle"`  

Editing  

`/Applications/OrcaSlicer.app/Contents/Resources/profiles/Creality/filament/Creality Generic TPU.json`  

didn't work in my case.
